### ISSUE TITLE
Invalid thumbnail served to VLC

### ISSUE THREAD

**Issue by ytimenkov** [gerbera/gerbera#430]
Gerbera 1.3.0 with libupnp-1.8.4, VLC 3.0.6

Thumbnails are not shown in VLC.
Wireshark trace shows that they are downloaded, but contents is not valid jpeg.
First, VLC sends request with Content-Range which Gerbera seems to not like:\
```
GET /content/media/object_id/1545/res_id/1/rh/6/ext/file.jpg HTTP/1.1
Host: 192.168.0.135:49152
Accept: */*
Accept-Language: en_US
User-Agent: VLC/3.0.6 LibVLC/3.0.6
Range: bytes=0-

HTTP/1.1 416 Requested Range Not Satisfiable
SERVER: Linux/4.20.10-gentoo, UPnP/1.0, Portable SDK for UPnP devices/1.8.4
CONNECTION: close
CONTENT-LENGTH: 70
Accept-Ranges: bytes
CONTENT-TYPE: text/html

<html><body><h1>416 Requested Range Not Satisfiable</h1></body></html>
```

Then it repeats same URL with HTTP/1.0:
```
GET /content/media/object_id/1545/res_id/1/rh/6/ext/file.jpg HTTP/1.0
Host: 192.168.0.135:49152
User-Agent: VLC/3.0.6 LibVLC/3.0.6
Icy-MetaData: 1

HTTP/1.0 200 OK
CONTENT-TYPE: image/jpeg
DATE: Sat, 09 Mar 2019 16:58:00 GMT
LAST-MODIFIED: Sat, 06 Feb 2016 12:59:09 GMT
SERVER: Linux/4.20.10-gentoo, UPnP/1.0, Portable SDK for UPnP devices/1.8.4
X-User-Agent: redsonic
Content-Disposition: attachment; filename="VID_20180103_143716.mp4"


......JFIF
```

The problem is with an extra newline following response headers. It gets saved into file resulting in a corrupted jpeg:
```
00000000  0d 0a ff d8 ff e0 00 10  4a 46 49 46 00 01 01 00  |........JFIF....|
```

wget downloads same url without any problems. And as I understand HTTP/1.0 also requires only one empty line between response headers and body

**Comment by elmodaddyb** [comment]
We fixed a few header protocol issues in #316 & #320 .  This is likely a similar scenario.

I agree the spec is right, and gerbera protocol needs fixing.  Its troublesome that tested on PS3 & wget the content downloads OK, but not on more strict systems.

**Comment by elmodaddyb** [comment]
Hello,

I was able to troubleshoot this and found that it is due to the fact that the Thumbnail code in Gerbera does not set the content length when serving the image.

I sent a request into Gerbera as you specified in `#1`.  The reason you receive a `416`
is because the libupnp, cannot render out the length as per code here ---> [libupnp - webserver.c](https://github.com/mrjimenez/pupnp/blob/aae0c586042d32bbd6e680dcd89de6a35709715e/upnp/src/genlib/net/http/webserver.c#L812)

This is a downstream error from the root cause, being that Gerbera does not set the `Content-Length`  for the thumbnail image.

https://github.com/gerbera/gerbera/blob/master/src/file_request_handler.cc#L162

> The TODO is your error exactly.

When I set the content length manually (i.e. hard-coded) based on a known thumbnail, the `#1` request is successful.  If I remove the `Range` header it is also successful (as seen with wget, or PS3, etc).

I suspect the subsequent request for the attachment video is just fallout from prior bad request.
